code/datums/components/holderloving.dm

```diff
@@ -0,0 +1,69 @@
+/** Holder Loving Component
+ *
+ * This component is assigned to an [/obj/item], and also keeps track of a [holder].
+ * The [parent] is 'bound' to [holder]. [parent] will be kept either directly
+ * inside [holder], or in the inventory of a [/mob] that is itself holding [holder].
+ *
+ * If [parent] is placed in a [loc] that is not [holder] or [holder].[loc]
+ * (if it's a mob), it is placed back inside [holder].
+ *
+ * This is intended for items that are a 'part' of another item.
+ *
+ * It can also delete [parent] when [holder] is deleted.
+ *
+ */
+/datum/component/holderloving
+	can_transfer = TRUE
+	/// Item that parent is bound to
+	var/obj/item/holder
+	/// If parent is deleted when the holder gets deleted
+	var/del_parent_with_holder = FALSE
+
+/datum/component/holderloving/Initialize(holder, del_parent_with_holder)
+	if(!isitem(parent) || !holder)
+		return COMPONENT_INCOMPATIBLE
+	src.holder = holder
+	src.del_parent_with_holder = (del_parent_with_holder || FALSE)
```

Comment:
```suggestion
	src.del_parent_with_holder = del_parent_with_holder
```

You're not checking explicitly for `== FALSE` later anyway, so all your checks will work fine.

Reply:
I did this because I just don't want del_parent_with_holder to be null

Reply:
It doesn't matter, but if you want that then do `if (del_parent_with_holder) src.del_parent_with_holder = ...`, for consistency.